entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultEntitlement.java

```diff
@@ -538,7 +538,7 @@ private LocalDate getLocalDateFromEntitlementPolicy(final EntitlementActionPolic
             default:
                 throw new RuntimeException("Unsupported policy " + entitlementPolicy);
         }
-        return (cancellationDate.compareTo(getEffectiveStartDate()) < 0) ? getEffectiveStartDate() : cancellationDate;
+        return (cancellationDate.compareTo(internalTenantContext.toLocalDate(getEffectiveStartDate())) < 0) ? internalTenantContext.toLocalDate(getEffectiveStartDate()) : cancellationDate; //TODO_1375 - Return LocalDate for now. Revisit later to check if this method should return DateTime
```

Comment:
Same comment as above, should we change this method to return a `DateTime` instead of `LocalDate` ?